pkg/apis/kops/cluster.go

```diff
@@ -603,6 +607,8 @@ type ClusterSubnetSpec struct {
 	Type SubnetType `json:"type,omitempty"`
 	// PublicIP to attach to NatGateway
 	PublicIP string `json:"publicIP,omitempty"`
+	// define aws subnet number in decimanl
+	AwsIpv6SubnetNum int `json:"awsIpv6SubnetNum,omitempty"`
```

Comment:
any logic you have in mind to convert boolean into netnum number? cause I thought it first to do it based on the subnet list position but if someone changes the subnet position it will be screwed.
```
cidrsubnet(aws_vpc.spike-np-tech-in.ipv6_cidr_block, 8, 1(netnum))
...
cidrsubnet(aws_vpc.spike-np-tech-in.ipv6_cidr_block, 8, 2(netnum))
```

Reply:
There would have to be a phase after the task actuals were constructed (determining which subnets were already in use) to then assign unused subnets to the expecteds that need them.

Reply:
This is an additional subnet that looks similar to what we have for additional network CIDRs and may be good to keep consistent.
https://github.com/kubernetes/kops/blob/b054fb37b7acbf164a072c428caf789b1b5c5e1e/k8s/crds/kops.k8s.io_clusters.yaml#L41-L48

Reply:
The issue here is that you don't know the subnet's full prefix until AWS assigns a `/56` to the VPC. The information needed is which of the 256 `/64`s of the `/56` to give to the subnet.

Reply:
@johngmyers  we can take this field as boolean but don't you think taking decimal gives more flexibility as people can group subnets like for all kops cluster they can use 0-128 for more secure public subnet and 129-255 for another public subnet and similar groups like this and as we are also taking subnet ipv4 CIDR so it will be same as that.

Reply:
Not all subnets in a VPC is owned by kOps, so I think we have to specify the number.